cmd/celestia-appd/cmd/start.go

```diff
@@ -232,6 +232,107 @@ func startStandAlone(ctx *server.Context, appCreator srvrtypes.AppCreator) error
 		return err
 	}
 
+	// Add the tx service to the gRPC router. We only need to register this
+	// service if API or gRPC is enabled.
+	if config.API.Enable || config.GRPC.Enable {
+		app.RegisterTxService(clientCtx)
+	}
+
+	metrics, err := startTelemetry(config)
+	if err != nil {
+		return err
+	}
+
+	var apiSrv *api.Server
+	if config.API.Enable {
+		genDocProvider := node.DefaultGenesisDocProviderFunc(ctx.Config)
+		genDoc, err := genDocProvider()
+		if err != nil {
+			return err
+		}
+
+		clientCtx = clientCtx.WithHomeDir(home).WithChainID(genDoc.ChainID)
+
+		if config.GRPC.Enable {
+			_, _, err := net.SplitHostPort(config.GRPC.Address)
+			if err != nil {
+				return err
+			}
+
+			maxSendMsgSize := config.GRPC.MaxSendMsgSize
+			if maxSendMsgSize == 0 {
+				maxSendMsgSize = serverconfig.DefaultGRPCMaxSendMsgSize
+			}
+
+			maxRecvMsgSize := config.GRPC.MaxRecvMsgSize
+			if maxRecvMsgSize == 0 {
+				maxRecvMsgSize = serverconfig.DefaultGRPCMaxRecvMsgSize
+			}
+
+			// If grpc is enabled, configure grpc client for grpc gateway.
+			grpcClient, err := grpc.NewClient(
+				config.GRPC.Address,
+				grpc.WithTransportCredentials(insecure.NewCredentials()),
+				grpc.WithDefaultCallOptions(
+					grpc.ForceCodec(codec.NewProtoCodec(clientCtx.InterfaceRegistry).GRPCCodec()),
+					grpc.MaxCallRecvMsgSize(maxRecvMsgSize),
+					grpc.MaxCallSendMsgSize(maxSendMsgSize),
+				),
+			)
+			if err != nil {
+				return err
+			}
+
+			clientCtx = clientCtx.WithGRPCClient(grpcClient)
+			ctx.Logger.Debug("grpc client assigned to client context", "target", config.GRPC.Address)
+		}
```

Comment:
[nit] is it possible to refactor this block so that there's less layers of indentation? Something like:

```go
if config.GRPC.Enable {
// setup GRPC
}

if config.API.Enable {
// setup API
}
```

Currently, there are two `if config.GRPC.Enable` blocks which seems repetitive and not immediately obvious why one needs to occur inside the `if.config.API.Enable` block